torch/csrc/distributed/c10d/FlightRecorder.hpp

```diff
@@ -1,16 +1,14 @@
 #pragma once
-
-// TODO: Make Fligth Recorder device agnostic
-#ifdef USE_C10D_NCCL
-
 #include <cstdio>
 #include <cstdlib>
 
 #include <memory>
 #include <mutex>
 
 #include <ATen/ATen.h>
+#ifdef USE_C10D_NCCL
 #include <ATen/cuda/CUDAEvent.h>
+#endif // USE_C10D_NCCL
```

Comment:
The only place that uses `CUDAEvent` is `getDurationFromEvent` function. Shall we template this function and remove `#include <ATen/cuda/CUDAEvent.h>` entirely?